File: yep/src/main/java/catchla/yep/activity/SkillUsersActivity.java
package catchla.yep.activity;

import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import catchla.yep.Constants;
import catchla.yep.R;
import catchla.yep.adapter.TabsAdapter;
import catchla.yep.fragment.DiscoverFragment;
import catchla.yep.graphic.EmptyDrawable;
import catchla.yep.model.Skill;
import catchla.yep.util.ThemeUtils;
import catchla.yep.util.Utils;
import catchla.yep.view.TabPagerIndicator;

/**
 * Created by mariotaku on 15/6/2.
 */
public class SkillUsersActivity extends SwipeBackContentActivity implements Constants {

    private ViewPager mViewPager;
    private TabPagerIndicator mPagerTab;
    private View mPagerOverlay;

    private TabsAdapter mPagerAdapter;

    @Override
    public void onContentChanged() {
        super.onContentChanged();
        mViewPager = (ViewPager) findViewById(R.id.view_pager);
        mPagerTab = (TabPagerIndicator) findViewById(R.id.view_pager_tabs);
        mPagerOverlay = findViewById(R.id.pager_window_overlay);
    }

    @Override
    public boolean onCreateOptionsMenu(final Menu menu) {
        getMenuInflater().inflate(R.menu.menu_skill_users, menu);
        return true;
    }

    @Override
    public boolean onPrepareOptionsMenu(final Menu menu) {
        final MenuItem itemSkill = menu.findItem(R.id.add_skill);
        final boolean hasSkill = Utils.hasSkill(Utils.getAccountUser(this, getAccount()), getSkill());
        itemSkill.setVisible(!hasSkill);
        itemSkill.setEnabled(!hasSkill);
        return true;
    }

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_skill);
        mPagerAdapter = new TabsAdapter(this, getSupportFragmentManager());
        mViewPager.setAdapter(mPagerAdapter);
        mPagerTab.setViewPager(mViewPager);
        mPagerTab.updateAppearance();

        final Skill skill = getSkill();
        displaySkill(skill);
        final Bundle masterArgs = new Bundle();
        masterArgs.putParcelable(EXTRA_ACCOUNT, getAccount());
        masterArgs.putStringArray(EXTRA_MASTER, new String[]{skill.getId()});
        final Bundle learningArgs = new Bundle();
        learningArgs.putParcelable(EXTRA_ACCOUNT, getAccount());
        learningArgs.putStringArray(EXTRA_LEARNING, new String[]{skill.getId()});
        mPagerAdapter.addTab(DiscoverFragment.class, getString(R.string.master), 0, masterArgs);
        mPagerAdapter.addTab(DiscoverFragment.class, getString(R.string.learning), 0, learningArgs);


        ThemeUtils.initPagerIndicatorAsActionBarTab(this, mPagerTab, mPagerOverlay);
        ThemeUtils.setCompatToolbarOverlay(this, new EmptyDrawable());
        ThemeUtils.setCompatContentViewOverlay(this, new EmptyDrawable());
    }

    private Skill getSkill() {
        return getIntent().getParcelableExtra(EXTRA_SKILL);
    }

    private void displaySkill(final Skill skill) {
        setTitle(skill.getNameString());
    }

}

File: src/main/java/catchla/yep/activity/SkillUsersActivity.kt
package catchla.yep.activity

import android.os.Bundle
import android.support.v4.view.ViewPager
import android.view.Menu
import android.view.View
import catchla.yep.Constants
import catchla.yep.R
import catchla.yep.adapter.TabsAdapter
import catchla.yep.fragment.DiscoverFragment
import catchla.yep.graphic.EmptyDrawable
import catchla.yep.model.Skill
import catchla.yep.util.ThemeUtils
import catchla.yep.util.Utils
import org.mariotaku.ktextension.setItemAvailability
import catchla.yep.view.TabPagerIndicator

/**
 * Created by mariotaku on 15/6/2.
 */
class SkillUsersActivity : SwipeBackContentActivity(), Constants {

    private var mViewPager: ViewPager? = null
    private var mPagerTab: TabPagerIndicator? = null
    private var mPagerOverlay: View? = null

    private var mPagerAdapter: TabsAdapter? = null

    override fun onContentChanged() {
        super.onContentChanged()
        mViewPager = findViewById(R.id.view_pager) as ViewPager?
        mPagerTab = findViewById(R.id.view_pager_tabs) as TabPagerIndicator?
        mPagerOverlay = findViewById(R.id.pager_window_overlay)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_skill_users, menu)
        return true
    }

    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
        val hasSkill = Utils.hasSkill(Utils.getAccountUser(this, account), skill)
        menu.setItemAvailability(R.id.add_skill, !hasSkill)
        return true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_skill)
        mPagerAdapter = TabsAdapter(this, supportFragmentManager)
        mViewPager!!.adapter = mPagerAdapter
        mPagerTab!!.setViewPager(mViewPager)
        mPagerTab!!.updateAppearance()

        val skill = skill
        displaySkill(skill)
        val masterArgs = Bundle()
        masterArgs.putParcelable(Constants.EXTRA_ACCOUNT, account)
        masterArgs.putStringArray(Constants.EXTRA_MASTER, arrayOf(skill.id))
        val learningArgs = Bundle()
        learningArgs.putParcelable(Constants.EXTRA_ACCOUNT, account)
        learningArgs.putStringArray(Constants.EXTRA_LEARNING, arrayOf(skill.id))
        mPagerAdapter!!.addTab(DiscoverFragment::class.java, getString(R.string.master), 0, masterArgs)
        mPagerAdapter!!.addTab(DiscoverFragment::class.java, getString(R.string.learning), 0, learningArgs)


        ThemeUtils.initPagerIndicatorAsActionBarTab(this, mPagerTab, mPagerOverlay)
        ThemeUtils.setCompatToolbarOverlay(this, EmptyDrawable())
        ThemeUtils.setCompatContentViewOverlay(this, EmptyDrawable())
    }

    private val skill: Skill
        get() = intent.getParcelableExtra<Skill>(Constants.EXTRA_SKILL)

    private fun displaySkill(skill: Skill) {
        title = skill.nameString
    }

}
